oracles/src/repositories/ipfs_repository.py

```diff
@@ -0,0 +1,43 @@
+import aiohttp
+import settings
+from typing import Union
+
+NFT_STORAGE_LINK_BASE = "https://{}.ipfs.nftstorage.link"
+
+
+class IpfsRepository:
+    async def read_file(self, cid: str) -> Union[str, bytes]:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(NFT_STORAGE_LINK_BASE.format(cid)) as response:
+                response.raise_for_status()
+                try:
+                    return await response.text()
```

Comment:
We really need some max file size checks here? Ideally imo read file in chunks and max file size exceeded raise error?

Reply:
Added.